### ISSUE TITLE
Non-square payoff matrix

### ISSUE THREAD

**Issue by BartMassey** [BartMassey/mixed-strategies-rs#4]
The code has never been tested with a non-square payoff matrix. This should be done, and the test should be added to the test suite. I don't even understand how the inactive strategy naming thing works for a rectangular matrix, so there's that: I think some code probably needs to be added or changed somewhere.

**Comment by vi** [comment]
Indeed there seem to be issues:

```
$ printf '0 1 1\n0 -1 0\n' | mixed-strategies 
value 0.000
max 0:1.000 1:0.000
min 0:1.000 1:0.000 2:0.000

$ printf '0 1 1\n0 +1 0\n' | mixed-strategies 
thread 'main' panicked at 'index out of bounds: the len is 2 but the index is 2', /rustc/521d78407471cb78e9bbf47160f6aa23047ac499/src/libcore/slice/mod.rs:2721:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

**Comment by BartMassey** [comment]
Yes. The algorithm given in *Compleat Strategyst* really doesn't make clear how to handle this case, so I'm going to have to figure it out.

If anyone reading Chapter 6 understands how to update the active strategies in step 5 for an asymmetric matrix, I'd appreciate the help.

The obvious plan is to pad the schema out with inactive strategies to make it square, but this seems gross and I can't believe there isn't a better plan…

**Comment by BartMassey** [comment]
In the meantime, I've gotten rid of the panic by changing the solution format (commit 4e87c8d). The new format is probably better anyway, as it provides information about which strategies are active. But the answers for non-square schema are still wrong/buggy. Won't upload to `crates.io` until this bug is fixed, sadly.
